backend/stakepoold/server.go

```diff
@@ -108,7 +106,11 @@ func deriveChildAddresses(key *hdkeychain.ExtendedKey, startIndex, count uint32,
 		if err != nil {
 			return nil, err
 		}
-		addr, err := child.Address(params)
+		ecPubKey, err := child.ECPubKey()
+		if err != nil {
+			return nil, err
+		}
+		addr, err := dcrutil.NewAddressPubKeyHash(dcrutil.Hash160(ecPubKey.Serialize()), params, dcrec.STEcdsaSecp256k1)
```

Comment:
Need to ensure that this change has not affected any functionality.

Reply:
Also I noticed this lengthy line is duplicated three times. It be worth extracting it into a helper function.

Reply:
`deriveChildAddresses` could use a unit test.

Reply:
It's one of the few things that does 

https://github.com/decred/dcrstakepool/blob/347de5e15920055d899e17be3973ef7de40333e3/backend/stakepoold/server_test.go#L20

So, I guess this is producing the same values as before

Reply:
Making a test tho

Reply:
I think I found the right way to do this.

Reply:
Dave had informed me that the previous way was better, I will revert this
https://matrix.to/#/!HEeJkbPRpAqgAwhXWO:decred.org/$15702398494445ukKtC:decred.org?via=decred.org&via=matrix.org&via=pintobyte.com